yt/data_objects/static_output.py

```diff
@@ -1,4 +1,5 @@
 import abc
+import contextlib
```

Comment:
Following the stdlib's dominant import style for any given module is preferred (for instance we use `import os; os.path.....` and never `from os.path import`).
In this case, following the docs, it's seems more conventional to write `from contextlib import contextmanager`. I also see that you're not using anything else from the namespace.

```suggestion
from contextlib import contextmanager
```

To avoid creating noise, I won't make the 15-ish related suggestions to change this throughout the PR.

also: I'm aware that the way we use `abc` currently contradicts this rule of thumb.